prometheus/collectors/go_collector_latest_test.go

```diff
@@ -40,9 +44,331 @@ func TestGoCollectorMarshalling(t *testing.T) {
 		t.Errorf("json marshalling shoud not fail, %v", err)
 	}
 }
+func TestWithGoCollectorMemStatsMetricsDisabled(t *testing.T) {
```

Comment:
I think we don't need that. This is thoroughly tested in https://github.com/prometheus/client_golang/blob/b94d7e25292465ac578c2e885e6d61e03a81ec4b/prometheus/go_collector_latest_test.go#L77